### PR TITLE
razergenie: init at 0.8.1

### PR DESCRIPTION
**Pull request by f4814**
###### Motivation for this change
Add razergenie, a frontend for the openrazer daemon. To the best of my knowledge there is no openrazer frontend in nixpkgs at the moment, which renders nixos `hardware.openrazer` functionality almost useless. I also added myself as contributor and notified the RazerGenie maintainers here: https://github.com/z3ntu/RazerGenie/issues/70

I attempted this here https://github.com/NixOS/nixpkgs/pull/83106 but screwed up my PR. So here is a new, clean PR.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by evanjs** [comment]
Related: https://github.com/NixOS/nixpkgs/pull/82609
But yes, nothing I've seen at the moment on nixpkgs.
Thanks for the PR!

**Comment by evanjs** [comment]
Anyway, functionality-wise, I can change basic effects, DPI, etc.

I cannot open the "Custom Editor" and image downloads are disabled, so the device thumbnails are not present, though I assume these are known issues.

**Comment by f4814** [comment]
The custom editor needs to be supported for your devices (mine are not). Same goes for device thumbnails. When I run it with my hardware it outputs `".png mapping for device 'Razer Goliathus' (PID 3073) missing."`. So both are issues we cannot really fix.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2470,6 +2470,12 @@
     githubId = 2817965;
     name = "f--t";
   };
+  f4814n = {
```
Comment (etu) [inline maintainers/maintainer-list.nix:2473]: This should be a separate commit

File: pkgs/applications/misc/razergenie/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+, enableExperimental ? false
+, includeMatrixDiscovery ? false
+}:
+
+let
+  version = "0.8.1";
+  pname = "razergenie";
+
+in stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
```
Comment (etu) [inline pkgs/applications/misc/razergenie/default.nix:12]: We usually declare pname and version as separate attributes, so this should create the same result:
```suggestion
  inherit pname version;
```

File: pkgs/applications/misc/razergenie/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+, wrapQtAppsHook
+, enableExperimental ? false
+, includeMatrixDiscovery ? false
+}:
+
+let
+  version = "0.8.1";
+  pname = "razergenie";
+
+in stdenv.mkDerivation rec {
```
Comment (etu) [inline pkgs/applications/misc/razergenie/default.nix:11]: `rec` not needed since no attribute inside of the set is referenced within the set
```suggestion
in stdenv.mkDerivation {
```

**Review by etu — CHANGES_REQUESTED** [review]
Some minor code changes and you should split up the addition to the maintainer list to it's own commit.

But other than that it looks good :)

**Comment by evanjs** [comment]
> From what I understand the custom editor needs to be supported for your devices (mine are not). Same goes for device thumbnails. When I run it with my hardware it outputs `".png mapping for device 'Razer Goliathus' (PID 3073) missing."`. So both are issues we cannot really fix.

Yup! That’s what I figured. 
Mamba TE and Huntsman Elite on my end. 
This would definitely be enough to help anybody that just wants to “turn off their lights” without dealing with the scripting side of things as documented [here](https://openrazer.github.io/#nixos).

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (f4814) [reply maintainers/maintainer-list.nix:2473]: Done!

**Comment by f4814** [comment]
Fixed the style issues and moved the changes to the maintainers list to another commit.

**Comment by f4814** [comment]
Any reason not to merge this?

**Comment by etu** [comment]
Mostly me forgetting about this PR :smile: 

It builds fine, if I run it it just shows a window about something missing:
image

Is this expected?

**Comment by f4814** [comment]
Well. You need to enable the Openrazer daemon (which provides the API razergenie uses). So if `hardware.openrazer.enable` is not set (assuming you are using nixos) this is entirely expected.
